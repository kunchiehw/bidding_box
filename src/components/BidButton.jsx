import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { ALL_SUITS } from '../utils/constants';
import { bidToString, suitColor } from '../utils/helpers';
import { levelValidator } from '../utils/validators';

const propTypes = {
  level: levelValidator.isRequired,
  suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  isDisabled: PropTypes.bool.isRequired,
  handleBidButtonClick: PropTypes.func.isRequired,
};

class BidButton extends PureComponent {
  render() {
    return (
      <Button
        className="bid-button"
        onClick={() => { this.props.handleBidButtonClick(this.props.level, this.props.suit); }}
        disabled={this.props.isDisabled}
        color={suitColor(this.props.suit)}
        size="small"
      >
        {bidToString(this.props.level, this.props.suit)}
      </Button>
    );
  }
}

BidButton.propTypes = propTypes;

export default BidButton;
