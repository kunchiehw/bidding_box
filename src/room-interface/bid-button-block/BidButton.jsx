import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { bidToString, suitColor } from '../helper-RoomInterface';
import { ALL_SUITS } from '../../util/util';

const propTypes = {
  // TODO: Might need a specific validator
  level: PropTypes.number.isRequired,
  suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  isDisabled: PropTypes.bool.isRequired,
  handleBidButtonClick: PropTypes.func.isRequired,
};

class BidButton extends PureComponent {
  render() {
    return (
      <Button
        className="bid-button"
        onClick={() => { this.props.handleBidButtonClick(this.props.level, this.props.suit); }}
        disabled={this.props.isDisabled}
        color={suitColor(this.props.suit)}
        size="small"
      >
        {bidToString(this.props.level, this.props.suit)}
      </Button>
    );
  }
}

BidButton.propTypes = propTypes;

export default BidButton;
