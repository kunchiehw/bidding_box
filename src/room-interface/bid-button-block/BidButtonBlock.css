.bid-button-block {

}

.bid-button-block .bid-button-row .bid-button {
  width: 64px;
  min-width: 64px;
  margin: 2px 3px;
}
