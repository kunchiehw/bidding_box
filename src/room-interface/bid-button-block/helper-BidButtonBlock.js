export function suitToNumber(suit) {
  switch (suit) {
    case 'NOTRUMPS':
      return 5;
    case 'SPADES':
      return 4;
    case 'HEARTS':
      return 3;
    case 'DIAMONDS':
      return 2;
    case 'CLUBS':
      return 1;
    /* istanbul ignore next */
    default:
      return 0;
  }
}

export function shouldStandardBidButtonDisabled(currentLevel, currentSuit, level, suit) {
  if (currentSuit === null || currentLevel < level) {
    return false;
  }
  if (currentLevel > level) {
    return true;
  }
  return (suitToNumber(currentSuit) >= suitToNumber(suit));
}
