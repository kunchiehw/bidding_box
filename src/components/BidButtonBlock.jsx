import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import BidButton from './BidButton';
import { POSSIBLE_LEVELS, STANDARD_SUITS } from '../utils/constants';
import { isStandardBidAllowed } from '../utils/bidHelpers';
import { levelValidator } from '../utils/validators';

const propTypes = {
  level: levelValidator.isRequired,
  suit: PropTypes.oneOf(STANDARD_SUITS.concat(['PASS'])).isRequired,
  shouldDoubleButtonDisabled: PropTypes.bool.isRequired,
  shouldRedoubleButtonDisabled: PropTypes.bool.isRequired,
  handleBidButtonClick: PropTypes.func.isRequired,
};

class BidButtonBlock extends PureComponent {
  render() {
    const standardSuitsRows = POSSIBLE_LEVELS.map((buttonLevel) => {
      const standardSuitsButtons = STANDARD_SUITS.map(buttonSuit => (
        <Grid.Column key={buttonSuit} width={3} streched>
          <BidButton
            level={buttonLevel}
            suit={buttonSuit}
            isDisabled={isStandardBidAllowed(this.props.level, this.props.suit, buttonLevel, buttonSuit)}
            handleBidButtonClick={this.props.handleBidButtonClick}
          />
        </Grid.Column>
      ));
      return (
        <Grid.Row
          key={buttonLevel}
          className="bid-button-row standard-suit-bid-button-row"
          column={5}
          streched
        >
          {standardSuitsButtons}
        </Grid.Row>
      );
    });

    const specialSuitsRow = (
      <Grid.Row
        key={0}
        className="bid-button-row special-bid-button-row"
        column={3}
        streched
      >
        <Grid.Column key="PASS" width={5} streched>
          <BidButton
            level={0}
            suit="PASS"
            isDisabled={false}
            handleBidButtonClick={this.props.handleBidButtonClick}
          />
        </Grid.Column>
        <Grid.Column key="DOUBLE" width={5} streched>
          <BidButton
            level={0}
            suit="DOUBLE"
            isDisabled={this.props.shouldDoubleButtonDisabled}
            handleBidButtonClick={this.props.handleBidButtonClick}
          />
        </Grid.Column>
        <Grid.Column key="REDOUBLE" width={5} streched>
          <BidButton
            level={0}
            suit="REDOUBLE"
            isDisabled={this.props.shouldRedoubleButtonDisabled}
            handleBidButtonClick={this.props.handleBidButtonClick}
          />
        </Grid.Column>
      </Grid.Row>
    );

    return (
      <Grid className="bid-button-block">
        {standardSuitsRows}
        {specialSuitsRow}
      </Grid>
    );
  }
}

BidButtonBlock.propTypes = propTypes;

export default BidButtonBlock;
