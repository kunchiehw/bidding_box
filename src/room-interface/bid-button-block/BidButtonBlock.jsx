import React from 'react';
import PropTypes from 'prop-types';
import './BidButtonBlock.css';
import SuitBidButton from './SuitBidButton';
import OtherBidButton from './OtherBidButton';
import { SUITS, NON_SUITS } from '../../util/util';

const propTypes = {
  curBid: PropTypes.shape({
    level: PropTypes.number.isRequired,
    suit: PropTypes.oneOf(SUITS).isRequired,
  }),
  disabledDouble: PropTypes.bool.isRequired,
  disabledRedouble: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

const defaultProps = {
  curBid: null,
};

function BidButtonBlock(props) {
  const handleClick = (bid) => {
    props.handleClick(bid);
  };

  const levelPanels = [];

  for (let level = 1; level <= 7; level += 1) {
    const suitButtons = [];
    for (let suitIndex = 0; suitIndex < SUITS.length; suitIndex += 1) {
      const suit = SUITS[suitIndex];
      const bid = { level, suit };
      const suitButtonProp = {
        bid,
        curBid: props.curBid,
        handleClick,
      };
      suitButtons.push(<SuitBidButton key={suit} {...suitButtonProp} />);
    }
    levelPanels.push(<div key={level} className="bid-button-row suit-bid-button-row">{suitButtons}</div>);
  }

  const passButtonProp = {
    suit: NON_SUITS[0],
    isDisabled: false,
    handleClick,
  };

  const doubleButtonProp = {
    suit: NON_SUITS[1],
    isDisabled: props.disabledDouble,
    handleClick,
  };

  const redoubleButtonProp = {
    suit: NON_SUITS[2],
    isDisabled: props.disabledRedouble,
    handleClick,
  };

  return (
    <div className="bid-button-block">
      {levelPanels}
      <div className="bid-button-row other-bid-button-row">
        <OtherBidButton {...passButtonProp} />
        <OtherBidButton {...doubleButtonProp} />
        <OtherBidButton {...redoubleButtonProp} />
      </div>
    </div>
  );
}

BidButtonBlock.propTypes = propTypes;
BidButtonBlock.defaultProps = defaultProps;

export default BidButtonBlock;
