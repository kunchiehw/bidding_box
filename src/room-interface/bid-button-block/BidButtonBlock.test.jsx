import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BidButton from './BidButton';
import BidButtonBlock from './BidButtonBlock';
import { shouldStandardBidButtonDisabled } from './helper-BidButtonBlock';

it('helper shouldStandardBidButtonDisabled', () => {
  expect(shouldStandardBidButtonDisabled(0, 'PASS', 1, 'CLUBS')).toEqual(false);
  expect(shouldStandardBidButtonDisabled(2, 'SPADES', 2, 'NOTRUMPS')).toEqual(false);
  expect(shouldStandardBidButtonDisabled(2, 'NOTRUMPS', 3, 'SPADES')).toEqual(false);
  expect(shouldStandardBidButtonDisabled(4, 'HEARTS', 4, 'DIAMONDS')).toEqual(true);
  expect(shouldStandardBidButtonDisabled(5, 'DIAMONDS', 4, 'HEARTS')).toEqual(true);
});

it('BidButton render', () => {
  const bidButton1 = shallow(<BidButton
    level={0}
    suit="PASS"
    isDisabled={false}
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton1)).toMatchSnapshot();

  const bidButton2 = shallow(<BidButton
    level={0}
    suit="DOUBLE"
    isDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton2)).toMatchSnapshot();

  const bidButton3 = shallow(<BidButton
    level={0}
    suit="REDOUBLE"
    isDisabled={false}
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton3)).toMatchSnapshot();

  const bidButton4 = shallow(<BidButton
    level={1}
    suit="CLUBS"
    isDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton4)).toMatchSnapshot();

  const bidButton5 = shallow(<BidButton
    level={2}
    suit="DIAMONDS"
    isDisabled={false}
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton5)).toMatchSnapshot();

  const bidButton6 = shallow(<BidButton
    level={3}
    suit="HEARTS"
    isDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton6)).toMatchSnapshot();

  const bidButton7 = shallow(<BidButton
    level={5}
    suit="SPADES"
    isDisabled={false}
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton7)).toMatchSnapshot();

  const bidButton8 = shallow(<BidButton
    level={7}
    suit="NOTRUMPS"
    isDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButton8)).toMatchSnapshot();
});

it('BidButtonBlock render', () => {
  const bidButtonBlock1 = shallow(<BidButtonBlock
    currentLevel={0}
    currentSuit="PASS"
    shouldDoubleButtonDisabled
    shouldRedoubleButtonDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButtonBlock1)).toMatchSnapshot();

  const bidButtonBlock2 = shallow(<BidButtonBlock
    currentLevel={3}
    currentSuit="NOTRUMPS"
    shouldDoubleButtonDisabled={false}
    shouldRedoubleButtonDisabled
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButtonBlock2)).toMatchSnapshot();

  const bidButtonBlock3 = shallow(<BidButtonBlock
    currentLevel={7}
    currentSuit="HEARTS"
    shouldDoubleButtonDisabled
    shouldRedoubleButtonDisabled={false}
    handleBidButtonClick={() => null}
  />);
  expect(toJson(bidButtonBlock3)).toMatchSnapshot();
});
