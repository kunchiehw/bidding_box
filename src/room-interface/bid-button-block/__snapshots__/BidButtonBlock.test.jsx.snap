// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidButton render 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  Pass
</Button>
`;

exports[`BidButton render 2`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  X
</Button>
`;

exports[`BidButton render 3`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  XX
</Button>
`;

exports[`BidButton render 4`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 ♣
</Button>
`;

exports[`BidButton render 5`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 ♦
</Button>
`;

exports[`BidButton render 6`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 ♥
</Button>
`;

exports[`BidButton render 7`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 ♠
</Button>
`;

exports[`BidButton render 8`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 N
</Button>
`;

exports[`BidButtonBlock render 1`] = `
<div
  className="bid-button-block"
>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="1"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={1}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={1}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={1}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={1}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={1}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="2"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={2}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={2}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={2}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={2}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={2}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="3"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={3}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={3}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={3}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={3}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={3}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="4"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={4}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={4}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={4}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={4}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={4}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="5"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={5}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={5}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={5}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={5}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={5}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="6"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={6}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={6}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={6}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={6}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={6}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="7"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={7}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={7}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={7}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={7}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={7}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row special-bid-button-row"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      level={0}
      suit="PASS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      level={0}
      suit="DOUBLE"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      level={0}
      suit="REDOUBLE"
    />
  </div>
</div>
`;

exports[`BidButtonBlock render 2`] = `
<div
  className="bid-button-block"
>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="1"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={1}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={1}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={1}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={1}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={1}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="2"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={2}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={2}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={2}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={2}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={2}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="3"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={3}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={3}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={3}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={3}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={3}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="4"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={4}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={4}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={4}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={4}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={4}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="5"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={5}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={5}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={5}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={5}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={5}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="6"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={6}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={6}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={6}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={6}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={6}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="7"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="CLUBS"
      level={7}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="DIAMONDS"
      level={7}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="HEARTS"
      level={7}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={7}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={7}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row special-bid-button-row"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      level={0}
      suit="PASS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      level={0}
      suit="DOUBLE"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      level={0}
      suit="REDOUBLE"
    />
  </div>
</div>
`;

exports[`BidButtonBlock render 3`] = `
<div
  className="bid-button-block"
>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="1"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={1}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={1}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={1}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={1}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={1}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="2"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={2}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={2}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={2}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={2}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={2}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="3"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={3}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={3}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={3}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={3}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={3}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="4"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={4}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={4}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={4}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={4}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={4}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="5"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={5}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={5}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={5}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={5}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={5}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="6"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={6}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={6}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={6}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="SPADES"
      level={6}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="NOTRUMPS"
      level={6}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row standard-suit-bid-button-row"
    key="7"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="CLUBS"
      level={7}
      suit="CLUBS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="DIAMONDS"
      level={7}
      suit="DIAMONDS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      key="HEARTS"
      level={7}
      suit="HEARTS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="SPADES"
      level={7}
      suit="SPADES"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      key="NOTRUMPS"
      level={7}
      suit="NOTRUMPS"
    />
  </div>
  <div
    className="bid-button-row special-bid-button-row"
  >
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      level={0}
      suit="PASS"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={true}
      level={0}
      suit="DOUBLE"
    />
    <BidButton
      handleBidButtonClick={[Function]}
      isDisabled={false}
      level={0}
      suit="REDOUBLE"
    />
  </div>
</div>
`;
